### ISSUE TITLE
RWX Persistent storage / FSaaS integration

### ISSUE THREAD

**Issue by mhurtrel** [ovh/public-cloud-roadmap#21]
As a MKS user 
I want to be able to share persistent volumes across multiple worker nodes
to ensure compatibility with most demanding stateful scenarii

_Notes :_
We are working with our IaaS storage colleagues to support the upcoming FSaaS service as RWX Kubernetes storage.
This feature may be open in a selection of regions.
A workaround is to manually configure NAS-HA volumes https://docs.ovh.com/gb/en/kubernetes/Configuring-multi-attach-persistent-volumes-with-ovhcloud-nas-ha/

**Comment by mhurtrel** [comment]
This feature is linked to https://github.com/ovh/public-cloud-roadmap/issues/54

**Comment by RomainPhil** [comment]
Hello,
is this feature got a approximate delivery date for 2021 ?

the workaround is interesting but it would be a great way to make the cluster easyer to manage.

**Comment by mhurtrel** [comment]
Hi @RomainPhil . Yes : My storage colleagues target spring 2021. We will give more precise ETA as soon as we are able to :)

**Comment by jbod** [comment]
Hi @mhurtrel ,

The problem with the NAS-HA is that, appart from the storage, all resources are shared (RAM/CPU/bandwidth). See here: https://docs.ovh.com/gb/en/storage/faq-nas/

This can cause some troubles when using time-critical stateful apps.

As an example, I attempted to use a NFS volume for a PostgreSQL database (bitnami/postgres). When the database starts and launches the initialisation process, it takes so much time that Kubernetes believes there is a problem and kills the PostgreSQL container. This makes the initialisation process incomplete and leads to a non-working database. This can be solved by increasing the initialDelaySeconds value, but it's not a real good solution.

The only trick I found to make it working so far is to use Cinder storage and stick all the storage-requesting pods to a particular node using nodeSelector. But as we have more and more apps deployed, this makes too much services running on a single node.

I know it's a bad practice running databases in a k8s cluster, but I guess this is one of the stateful scenarii you imagined?

Do you think there is a way to allow a Cinder volume to be attached to multiple instances, as a temporary solution until FSaaS comes? Or maybe it is not possible and I'm not aware of that, sorry for asking then ;)

**Comment by mhurtrel** [comment]
Thanks for this detailed feedback !

I confirm that we work on 3 differents tracks:
- The most adequate one is FSaaS that is targeted for late spring
- A second one is multi attach cinder disks. I see various feedbacks from users of this approzch showing is can come with limitations if you have 2 workers nodes trying to write data at the same time so I anticipate this could be a problem with db.
- my nas colleagues are planning for nas with improved performance

Before this late Spring release ly best advice would be to keep this db outside ofbthe cluster, running it on vms for example, or using the workaround you suggest.

**Comment by ZuSe** [comment]
Hi @mhurtrel  is there any update on the issue. Spring is almost over.
Are you going to introduce a new storage class for it or will it work with csi-cinder-high-speed/classic out of the box?

I just run into the problem by some helm charts requiring RWX without much flexibility to make it work using RWO.

**Comment by mhurtrel** [comment]
Hello @ZuSe unfortunately my storage colleagues have delaued the fsaas for a couple of months, and I am waiting for a new eta whoch shoumd be somewhere during summer. I reasonably hope we will ve able to ibtegrate it late summer.

**Comment by mhurtrel** [comment]
The FSaaS this features rely on has been postponed to early 2022.  Sorry about the delay. The workaround described above still can help in the meantime.

**Comment by elavrom** [comment]
Hi @mhurtrel any news on this feature ? This looks to meet our needs and since we are in early 2022 I hoped that you could give a more precise ETA ? I don't think I should bother investigate the workaround if the feature is coming in a few months time.

**Comment by mhurtrel** [comment]
Hello. I dont have a new short term ETA unfortunaely but will work with my Storage colleagues to obtain this info ASAP

**Comment by matmicro** [comment]
It would be great if we could have some estimated timeframe.
Thanks

**Comment by mhurtrel** [comment]
Current ETA is late this calendar year

**Comment by larsonnn** [comment]
push

**Comment by mhurtrel** [comment]
Hello ! I confirm this is still a priority for us, and will come as soon as https://github.com/ovh/public-cloud-roadmap/issues/54 is made available by our storage colleagues. Sorry for the delay

**Comment by atino** [comment]
Is there a new ETA or is it actually coming this year? It would be a nice gift for christmas :)

**Comment by Beanux** [comment]
According to the last update of the closed ticket,
It's linked to #305 which was put in coming soon on their roadmap (~2 months, last week):
https://github.com/ovh/public-cloud-roadmap/projects/1#card-87623617

**Comment by bh42** [comment]
Hey there
#305 has been released and closed, any update regarding this issue?
Is it still blocked by #54 ?
Thanks

**Comment by mhurtrel** [comment]
Yes indeed, the dependancy is on #54

**Comment by zguig52** [comment]
After fighting some time, I found this project that do the job (creation of an internal NFS server that allow to share currently non-shareable PVCs from OVH via an Internal Storage Class using NFS):
https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner

**Comment by danvy** [comment]
> After fighting some time, I found this project that do the job (creation of an internal NFS server that allow to share currently non-shareable PVCs from OVH via an Internal Storage Class using NFS): https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner

Keep in mind that this quick and easy provisionner rely on a PVC that rely on a single disk. If this disk crashes, you'll loose your data. For a more robust by heavier solution, check out Rooks, OpenEBS or alternatives.

**Comment by gsontag** [comment]
It's been a very long time waiting for RWX, and NAS with NFS workaround is not a production grade solution. All project have to be rethink with the RWO constraints, it's a pain in the ass. Maybe it's time to put a highest priority on it and invest some energy.

**Comment by mhurtrel** [comment]
@gsontag it remains a top priority, and we have been making sure ou storage colleagues build a solution that allows cloud native use cases. They hace been dedicated multiple peopoe to this for a few quarters, and are looking for a release in the next semester hopefully (see #54 ) As soon as they release this cloud-native compatible filestorage, we will prioritize the integration to managed kubernetes as our main focus.

**Comment by monsty** [comment]
Also looking for ReadWriteMany compatibility...

**Comment by sevenGroupFrance** [comment]
Any news about this feature ?

**Comment by mhurtrel** [comment]
Hello @sevenGroupFrance I confirm our storage colleagues are actively developping the FSaaS prerequisites :
https://github.com/ovh/public-cloud-roadmap/issues/54 .

We raised our requirements and both teams have identified the associated effort (both on the FSaaS itslef and the CCM part), so that both will be adressed during spring. we epext a GA before summer and will update this as we approach the release.

**Comment by mperochon** [comment]
Hey, any new ETA for this feature ?

**Comment by robinComa** [comment]
still impossible to use ReadRightMany on the OVH offer after 4 years of waiting, do you have any news?

**Comment by matmicro** [comment]
> still impossible to use ReadRightMany on the OVH offer after 4 years of waiting, do you have any news?

Waiting this since before 2020... cannot wait anymore as we do not see real progress. We plan to move to AWS

**Comment by Llyth** [comment]
Hello @antonin-a,

Can we get an update, still waiting for any update, last informal news was last October.

**Comment by Rudedog9d** [comment]
This is certainly something I'm interested in! HA-NAS works for customers that can afford enterprise rates, but I represent several smaller organizations without the budget for $180+/mo  for storage. We're very excited for this feature, it's the single largest (possibly only) pain point of OVH managed K8S!

**Comment by brenault-g** [comment]
Any news ? The initial request was made five years ago...

**Comment by KeyPerformanceGroup** [comment]
I had a mail from OVH last week for an alpha on GRA5 !
I have complete their form and waiting for an answer...

**Comment by antonin-a** [comment]
Hello here, indeed we just launched an alpha phase for OVHcloud File Storage providing RWX storage class on MKS.
You can join the alpha here : https://labs.ovhcloud.com/en/file-storage/

**Limitations of the Alpha**
During the Alpha program, the service will come with the following limitations: 
- Limited availability in SBG5 (Strasbourg, France)
- No snapshots, no distant backups nor encryption
- Only available via the OpenStack API and Manilla CSI driver
- No downscale of storage volumes
- Basic CRUD features

Instructions on how to use the Manila (FSaaS) CSI on Kubernetes are available here : https://help.ovhcloud.com/csm/en-public-cloud-storage-file-storage-service-getting-started?id=kb_article_view&sysparm_article=KB0072892#instructions

**Comment by martinovh** [comment]
## File Storage BETA — now available on OVHcloud Public Cloud

What's new vs. the Alpha: -> more information available [here](https://labs.ovhcloud.com/en/file-storage/)!
- More data centers available: all GRA data centers and DE1 (SBG5 is still available)
- No need to register: create your first share via OVHcloud API V6, Manilla CSI driver, Openstack API (Control panel experience coming in February)
- Update in minimum volume size: starting at 150GiB to 10TiB

Starting today, create your first share for FREE (documentation available [here](https://help.ovhcloud.com/csm/en-public-cloud-storage-file-storage-service-getting-started?id=kb_article_view&sysparm_article=KB0072892)!

**NB:** The BETA phase is free for creating and using shared volumes. However, all created resources will be billed (€145/TiB/month with hourly billing) starting from the General Availability (GA) release in mid-March 2026. This pricing level is not a contractual commitment from OVHcloud, and GA prices are still subject to change.

---

### Reminder of the key features:


- Instantly increase your share size at the GiB level
- Create Share's Snapshots
- Linear performance allocation (if a volume is resized, performance is updated in consequence): 24 IOPS /GiB and 0.25 MB/s /GiB up to 16 000 IOPS and 128 MB/s.
- Solution based on OpenStack Manila and is compatible with the Manila CSI, Private Network compatibility (Vrack) and NFSv3 support
- Available in all GRA data centers, DE1 and SBG5 

**NB:** limitations will apply to the BETA, learn more [here](https://labs.ovhcloud.com/en/file-storage/). 

---

We’d love and need to hear your feedback on this iteration.  
Join the discussion on [**Discord**](https://discord.com/channels/850031577277792286/1153361284738326648) — your input is key to shaping the upcoming phases.
